src/node/db/API.js

```diff
@@ -298,15 +298,93 @@ exports.setText = function(padID, text, callback)
   getPadSafe(padID, true, function(err, pad)
   {
     if(ERR(err, callback)) return;
-    
+
     //set the text
     pad.setText(text);
-    
+
     //update the clients on the pad
     padMessageHandler.updatePadClients(pad, callback);
   });
 }
 
+exports.insertText = function(padID, newText, lineNum, authorMapper, callback)
+{
+  //text is required
+  if(typeof newText != "string")
+  {
+    callback(new customError("text is no string","apierror"));
+    return;
+  }
+
+  //check if lineNum is a number
+  if(!isNaN(parseInt(lineNum)))
+  {
+    lineNum = parseInt(lineNum);
+  }
+  else
+  {
+    callback(new customError("lineNum not given","apierror"));
+    return;
+  }
+
+  if(typeof authorMapper != "string")
+  {
+    callback(new customError("text is no authorMapper","apierror"));
+    return;
+  }
+
+  var authorMangager = require("./AuthorManager");
+  // Check author exists
+  authorMangager.createAuthorIfNotExistsFor(authorMapper, false, function(err, authorID)
+  {
+    if(ERR(err, callback)) return;
+
+    //author does not exist
+    if(authorID == false)
```

Comment:
Why do you check the authorID? As far as I can see there is no case that the authorID can be 'false'